airflow/dags/unzip_and_validate_gtfs_schedule/unzip_gtfs_schedule.py

```diff
@@ -0,0 +1,221 @@
+    we return the contents of the top level or the unique directory as the feed.
+    If there's a directory *and* other files at top level within zip, or if
+    there are nested directories, say we can't parse."""
+
+    logging.info(f"Processing {extract.name}")
+    zipfile_md5_hash = ""
+    try:
+        with fs.open(extract.path) as f:
+            zipfile_md5_hash = f.info()["md5Hash"]
+            zip = zipfile.ZipFile(f)
+        files, directories = summarize_zip_contents(zip)
+        logging.info(f"Found files: {files} and directories: {directories}")
+    except Exception as e:
+        return GTFSScheduleFeedExtractUnzipOutcome(
+            success=False,
+            extract_md5hash=zipfile_md5_hash,
+            extract_path=extract.path,
+            exception=e,
+        )
+    # more than one directory --> invalid zip
+    if len(directories) > 1:
+        logging.info(f"Invalid zip {extract.path}: Multiple directories found")
+        return GTFSScheduleFeedExtractUnzipOutcome(
+            success=False,
+            extract_md5hash=zipfile_md5_hash,
+            extract_path=extract.path,
+            zipfile_files=files,
+            zipfile_dirs=directories,
+        )
+    # mix of directories and files at top level --> invalid zip
+    elif (
+        len(directories) > 0
+        and len([item for item in zipfile.Path(zip, at="").iterdir()]) > 1
+    ):
+        logging.info(f"Invalid zip {extract.path}: Mix of directories and files found")
+        return GTFSScheduleFeedExtractUnzipOutcome(
+            success=False,
+            extract_md5hash=zipfile_md5_hash,
+            extract_path=extract.path,
+            zipfile_files=files,
+            zipfile_dirs=directories,
+        )
+    # if the only item in the zipfile is a directory at the root level
+    # we can treat the contents of that directory as if they were the feed
+    elif [item.is_dir() for item in zipfile.Path(zip, at="").iterdir()] == [True]:
+        logging.info(f"Valid zip {extract.path}: One toplevel directory found")
+        feed_directory = directories[0]
+        zipfile_files = process_feed_files(fs, extract, feed_directory)
+        return GTFSScheduleFeedExtractUnzipOutcome(
+            success=True,
+            extract_md5hash=zipfile_md5_hash,
+            extract_path=extract.path,
+            zipfile_files=files,
+            zipfile_dirs=directories,
+            extracted_files=[file.path for file in zipfile_files],
+        )
+    else:
+        logging.info(f"Valid zip {extract.path}")
+        zipfile_files = process_feed_files(fs, extract)
+        return GTFSScheduleFeedExtractUnzipOutcome(
+            success=True,
+            extract_md5hash=zipfile_md5_hash,
+            extract_path=extract.path,
+            zipfile_files=files,
+            zipfile_dirs=directories,
+            extracted_files=[file.path for file in zipfile_files],
+        )
+
+
+def unzip_extracts(day: pendulum.datetime):
```

Comment:
nit: would it make sense for `day` to be some date type rather than a datetime that is then truncated to the date?

Reply:
I was just copying this: https://github.com/cal-itp/data-infra/blob/main/jobs/gtfs-schedule-validator/gtfs_schedule_validator.py#L167-L173.... Is there a reason that was done that way? Can change. Also I thought that `execution_date` is like a timestamp anyway.

Reply:
ok, I think I'll leave it for now